### ISSUE TITLE
[BUG] Sound distorted internal player

### ISSUE THREAD

**Issue by jpd1968** [robvdpol/RaceControl#353]
**Describe the bug**
When playing a feed in the internal player the sound is distorted. Playback in MPV or VLC gives good sound.

**To Reproduce**
1. Go to a feed
2. Play stream World Feed of any other 

**Expected behavior**
A nice sound without distortion

**Desktop (please complete the following information):**
 - Windows 10 21H2 versie19044.1586
 - System type (64 bit)
 - Version 2.60

**F1TV account**
- Subscription type (Pro)
- Country NL
- Using VPN or proxy? Nope

**Comment by robvdpol** [comment]
@SuRGeoNix any ideas?

**Comment by SuRGeoNix** [comment]
@robvdpol Not really, nobody complained so far for similar issue. As you use the same feeds and you don't have issues I guess it's a hardware soundcard issue.

@jpd1968 Can you please provide more information and a log file? Do you have more than one soundcards? Does distorted mean slower or faster sample rate (Hz)? Does it happen all the time or with some specific feeds and audio tracks?

**Comment by jpd1968** [comment]
I'll run some tests and provide the logs. The answers to your questions  as
far I can tell from my workstation, which is not the device RC is installed
on. Further on I will install it on another device and circle com back to
you.

*Can you please provide more information and a log file?*I'll run some
tests on different hardware and sent the results including logs.
*Do you have more than one soundcards?*
Not that I know of it's a HP Pavillion laptop, but I'll  check it.
* Does distorted mean slower or faster sample rate (Hz)?*
It means disturbed(?) as in not sounding right and with stutter, while the
video is perfect. It's has a certain rythm.
*Does it happen all the time or with some specific feeds and audio tracks?*
I've tried world feed, Driver Cam and Pit Lane Channel (sorry for the old
names ;-) ) and several audio tracks. Also live and replays.


Op zo 20 mrt. 2022 om 08:15 schreef John Stamatakis <
***@***.***>:

**Comment by jpd1968** [comment]
[RaceControl.log](https://github.com/robvdpol/RaceControl/files/8311026/RaceControl.log)
[Laptop_Spec.txt](https://github.com/robvdpol/RaceControl/files/8311050/Laptop_Spec.txt)

It is definitely device specific, on my Game PC I've no problems. On my laptop (see specs and log) it is still stuttering audio. Live and replay. Only the internal player. See no errors in the log. See example video. 

https://user-images.githubusercontent.com/94355529/159158727-25f157b0-50b3-4ac6-b9e0-cf587920cea6.mp4

**Comment by mbrouwer** [comment]
I have the exact same problem but only in the following conditions:
- External display (TV) in "Extend Desktop" projection mode
- When the screen is dragged to the external display
Disabling some optimizations in the TV fixed this but the quality is less, using the HDMI in on my receiver fixed it also, but I feel I dont have the highest video either.

The reason I wanted this is so I can also listen to GrandPrixRadio on my speakers and not on my laptop so its not a very big problem anyways.

Anyways, I have chucked it up to my external display (TV) as  all HDMI ports have it but my receiver does not have it. It might be a bug in the app but from a development point of view that's almost impossible to debug without me sending my TV to @robvdpol :)

**Comment by SuRGeoNix** [comment]
@robvdpol Can you give the right direction to @jpd1968 to provide us with the Flyleaf log (if possible)?

@jpd1968 The problem seems to be with the GPU, maybe you have notice also frame drops? Your laptop has two GPU adapters and it's possible for some reason that it uses the 'wrong' one which is not able to perform hardware decoding and causes the audio issue.

**Comment by jpd1968** [comment]
[Flyleaf.log](https://github.com/robvdpol/RaceControl/files/8311362/Flyleaf.log)
Yoe mean this, its from 2.6.0? @SuRGeoNix 

Tried both adapters, assigned them with the Nvidia settings but the result was the same.

**Comment by SuRGeoNix** [comment]
@jpd1968 Yes that one but it misses the initialization. It's clear though that it drops also video frames which means that it probably uses the intel GPU adapter which can't handle the video playback properly.

**Comment by jpd1968** [comment]
All right, but when a assign the Nvidia adapter the sound problem still
exists, for now I'll us the MVP player. If you want more information pleas
ask.

Op zo 20 mrt. 2022 om 15:02 schreef John Stamatakis <
***@***.***>:

**Comment by SuRGeoNix** [comment]
@jpd1968 How do you assign the NVidia? You need to see the logs to ensure which card Flyleaf uses. I'm quite sure it uses the slow one.

**Comment by jpd1968** [comment]
In the Nvidia settings you can assign the program to integrated graphics or
the Nvidia. But you're right I can't confirm it.

Op zo 20 mrt. 2022 15:06 schreef John Stamatakis ***@***.***>:

**Comment by jpd1968** [comment]
[Flyleaf.log](https://github.com/robvdpol/RaceControl/files/8311414/Flyleaf.log)
I disabled the Intel GPU same result. Use 2.6.0 so here is another flyleaf log.

**Comment by SuRGeoNix** [comment]
@jpd1968 Can't see the details because of the Flyleaf's log level. However is clear this line

```
03.25.20.021 | Error | [#1] [Renderer      ] D3D11 Video Processor Initialization Failed
```
Which means that probably uses the intel and fails to activate the video/hardware acceleration. I have no idea why by default directx returns this GPU. I will try to resolve it with the next updates but ideally @robvdpol could expose Flyleaf.Config.xml for more advanced configuration such as this one which could force the other GPU adapter.

**Comment by SuRGeoNix** [comment]
@mbrouwer Feel free to open a new issue (even directly in [Flyleaf](https://github.com/SuRGeoNix/Flyleaf)). I'm not sure how by disabling some optimizations in the TV fixes it (what kind of optimizations? HDR?). Are other players work properly, might your GPU can't handle a 4K TV?

**Comment by mbrouwer** [comment]
> @mbrouwer Feel free to open a new issue (even directly in [Flyleaf](https://github.com/SuRGeoNix/Flyleaf)). I'm not sure how by disabling some optimizations in the TV fixes it (what kind of optimizations? HDR?). Are other players work properly, might your GPU can't handle a 4K TV and by reducing the resolution fixes the issue?

Yeah all other players are working fine and its indeed the HDR/4K optimization which I can set per HDMI in. Tried to force the RaceControl app to not use the GPU but it still shows 26% GPU - Decode in task manager. Like I said, not a big issue for me as I'm not using the audio from the stream anyways and can always try to use another laptop.

**Comment by bkbartk** [comment]
I have the same issue while dragging the player to another screen of other location of the screen.
the audio starts stottering.
I changed the graphical driver from intel to Nvidea but it doesn't make any difference.

I just notice now, the issue does not occur if I use MPV instead of Watch

**Comment by SuRGeoNix** [comment]
@bkbartk This sounds like a different issue as it happens to you only when you drag it to different screen. Can you try set the primary screen maybe to the other monitor and see if that still happens? I will try to find a way to reproduce it and fix it in to the upcoming versions.

**Comment by jpd1968** [comment]
I rolled back to 2.5.5. and the issue does not exist there.

Op za 19 mrt. 2022 om 15:39 schreef Rob ***@***.***>:

**Comment by SuRGeoNix** [comment]
Hi @jpd1968, it is been a while and there are a lot of updates of Flyleaf not applied yet to the RaceControl. Let me know if you can test the latest [flyleaf](https://github.com/SuRGeoNix/Flyleaf/releases) version and if the issue still exists.

**Comment by jpd1968** [comment]
Hi,

I haven't used Racecontrol for a long time, because of the difficulties
with the authorisation, it never seems to work for me. I'll have a look
into flyleaf and try to get it working.

Regards,
Jeroen

Op wo 12 okt. 2022 om 13:19 schreef John Stamatakis <
***@***.***>:
